### PR TITLE
Upgrade TFX to 1.6.1

### PR DESCRIPTION
**Pull request by jwwwb**
## Describe changes
Upgrades TFX to version 1.6.1 and adds string instance check for pipeline_root which is now of Union type in this TFX version.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by jwwwb** [comment]
> @jwwwb I think the same `pipeline_root` attribute should also be used in the airflow orchestrator, do we need any additional checks there as well?

For some reason mypy doesn't complain about it in that file (on the others it thinks it's a `Union[str, Placeholder]` but not there). 🤷

File: src/zenml/integrations/airflow/orchestrators/airflow_dag_runner.py [hunk 1]
```diff
@@ -130,17 +130,16 @@ def run(
             is_paused_upon_creation=False,
             catchup=catchup,
         )
+
+        pipeline_root = tfx_pipeline.pipeline_info.pipeline_root
+
```
Comment (stefannica) [inline src/zenml/integrations/airflow/orchestrators/airflow_dag_runner.py:135]: How come you don't need to check the type here as you do everywhere else ?

```
        if not isinstance(pipeline_root, str):
            raise TypeError(
                "TFX Pipeline root may not be a Placeholder, "
                "but must be a specific string."
            )
```

**Review by stefannica — APPROVED** [review]
Had one question, but looks ok otherwise (haven't tested it)

File: src/zenml/integrations/airflow/orchestrators/airflow_dag_runner.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/integrations/airflow/orchestrators/airflow_dag_runner.py:135]: Nevermind, this has already been addressed before :)

**Review by bcdurak — APPROVED** [review]

**Pull request merged**